Drivers/Src/Switches.c

```diff
@@ -13,88 +16,107 @@
  * @return  None
  */ 
 void Switches_Init(void){
+    OS_ERR err;
+    OSMutexCreate(&CommMutex, "communications Mutex", &err);
+    assertOSError(0,err);
     BSP_SPI_Init();
     //Sets up pins 0-7 on GPIOA as input 
-    uint8_t initTxBuf[3]={SPI_OPCODE_R, SPI_IODIRA, 0x00};
-    uint8_t initRxBuf[2] = {0}; 
-    BSP_SPI_Write(initTxBuf,3);
-    do{
-        BSP_SPI_Read(initRxBuf, 2);
-    }while(initRxBuf[1] == SPI_IODIRA);
+    uint8_t initTxBuf[3]={SPI_OPCODE_R, SPI_IODIRA, 0};
+    uint8_t initRxBuf = 0;
+    BSP_GPIO_Write_Pin(PORTA, GPIO_Pin_4, OFF);
```

Comment:
PORTA, GPIO_Pin_4 should probably be replaced with a
`#define SPI_PORT PORTA`
`#define SPI_CS_PIN GPIO_Pin_4` (or something similar)
for readability